atmo/jobs/forms.py

```diff
@@ -118,13 +119,14 @@ def clean_notebook(self):
                                         'allowed to be uploaded')
         return notebook_file
 
-    def save(self):
+    def save(self, commit=True):
         # create the model without committing, since we haven't
         # set the required created_by field yet
         spark_job = super(BaseSparkJobForm, self).save(commit=False)
 
-        # actually save the scheduled Spark job, and return the model object
-        spark_job.save(self.cleaned_data['notebook'])
+        if commit:
```

Comment:
Ah, this is where this breaks apart a bit, assuming someone would call this BaseSparkJobForm.save with commit=False, the notebook would never be saved, even if the save would be committed after the fact.

A slightly better version to this I think is to move the call to `scheduling.spark_job_add` from the `SparkJob.save` model save method to the one in the form, so it's clear that passing in `commit=False` will have this effect and not miss the connection.